### DISCUSSION TITLE
Custom S3 Server

### DISCUSSION THREAD

**Discussion by l0nax** [asciinema/discussions#112] [Q&A]
Is there a possibility instead of using the Amazon S3 Server-Service using a self-hosted S3 Server (like [MinIO](https://github.com/minio/minio/))?

This would a nice feature :)

**Comment by ku1ik** [comment]
It's not possible at the moment, but it would be rather easy to implement.

We have some S3 config here https://github.com/asciinema/asciinema-server/blob/32c30f6785e23d89223fd2cf915da9d6f268b414/config/releases.exs#L28-L36

We'd need to implement support for host/port vars there.

**Comment by pratikbin** [comment]
Waiting for this feature

**Comment by imroc** [comment]
Not implemented yet?

**Comment by ku1ik** [comment]
Check out docs: https://docs.asciinema.org/manual/server/self-hosting/configuration/#file-store

The Cloudflare R2 example should be easy to adapt for MinIO.
